library/math/complex.lisp

```diff
@@ -24,8 +26,21 @@
 #+coalton-release
 (cl:declaim #.coalton-impl/settings:*coalton-optimize-library*)
 
+;;; Tell Coalton that we should expect Complex to have a special type representation.
+(cl:eval-when (:compile-toplevel :load-toplevel :execute)
+  (cl:setf (cl:gethash 'Complex coalton-impl/typechecker/lisp-type:*special-type-representations*)
+           (cl:lambda (ty)
+             ;; (cl:format cl:t "testing ~S~%" ty)
+             (cl:typecase ty
+               (ty:tycon
+                (cl:case (ty:tycon-name ty)
```

Comment:
This should look at the lisp type of it's argument so that newtypes are handled correctly.